dts/bindings/kscan/gpio-kscan.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2020, Pete Johanson
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO keyboard matrix controller
+
+compatible: "gpio-kscan"
+
+include: kscan.yaml
+
+properties:
+    row-gpios:
+        type: phandle-array
+        required: true
+    col-gpios:
+        type: phandle-array
+        required: true
+    debounce-period:
+        type: int
+        default: 5
```

Comment:
It would be better if you add a Kconfig entry for your debounce period in your Kconfig.gpio. You could change tune it for other keyboard brands.  See the way it was done in Kconfig.xec and ini my opinion this parameter shouldn't be part of the dts/bindings as this is more related to keyboards you can swap and test.  There are keyboard that do not have diodes as these feature makes them more expensive. Therefore, I also believe Diode-direction should be part of your kconfig, in fact you could add another flag to specify where your keyboard has diodes.  This is just.a suggestion.